### ISSUE TITLE
Add warnings when input has too much / too little data 

### ISSUE THREAD

**Issue by mastoffel** [alan-turing-institute/autoemulate#216]
* models might overfit when there are not enough datapoints relative to the number of variables. We should check good rules of thumb and give a warning if this is the case
* some models, in particular GP's, get very slow with too many samples. We should give a warning when the running time will be long. Even better, we could provide a drop in replacement for the `scikit-learn` GP that deals with larger datasets (probably at the cost of accuracy). This is an active area of research but there might be solutions.

Thanks to @MaxBalmus for the feedback.
